cookbook/workflows/manual_trigger/basic_stream_custom_execution_function_workflow.py

```diff
@@ -0,0 +1,338 @@
+    model=OpenAIChat(id="gpt-4o"),
+    tools=[DuckDuckGoTools()],
+    instructions="Extract key insights and content from blog posts",
+)
+
+content_planner = Agent(
+    name="Content Planner",
+    model=OpenAIChat(id="gpt-4o"),
+    instructions="Create engaging social media content plans based on analysis",
+)
+
+# Define research team for complex analysis
+research_team = Team(
+    name="Research Team",
+    mode="coordinate",
+    members=[blog_analyzer, content_planner],
+    instructions="Analyze content and create comprehensive social media strategy",
+)
+
+
+# Streaming-enabled custom execution functions
+def streaming_blog_analysis_function(
+    task_input: TaskInput,
+) -> Iterator[Union[RunResponseContentEvent, TaskOutput]]:
+    """
+    Streaming custom function that yields events during execution and returns TaskOutput
+    """
+    query = task_input.get_primary_input()
+
+    # Add custom context and formatting
+    enhanced_query = f"""
+        BLOG ANALYSIS REQUEST:
+        Topic: {query}
+        
+        Please provide:
+        1. Key themes and trends
+        2. Target audience insights  
+        3. Content opportunities
+        4. SEO considerations
+        
+        Additional context from previous tasks:
+        {task_input.previous_outputs if task_input.previous_outputs else "No previous context"}
+    """
+
+    # Call the agent with enhanced input
+    try:
+        accumulated_content = ""
+        response = None
+
+        # Stream the agent response
+        for event in blog_analyzer.run(enhanced_query, stream=True):
+            if hasattr(event, "content") and event.content:
+                accumulated_content += event.content
+                # Forward the streaming content
+                yield event
+            if hasattr(event, "response"):
+                response = event.response
+
+        # Custom postprocessing
+        enhanced_content = f"""
+## Custom Blog Analysis Results
+
+**Original Query:** {query}
+
+**Analysis:**
+{accumulated_content}
+
+**Custom Insights:**
+- Analysis completed with enhanced context
+- Ready for next stage processing
+- Confidence level: High
+        """.strip()
+
+        # Return final TaskOutput
+        yield TaskOutput(
+            content=enhanced_content,
+            response=response,
+            data={"analysis_type": "custom", "confidence": "high"},
+            metadata={
+                "function_name": "streaming_blog_analysis_function",
+                "preprocessing": True,
+                "postprocessing": True,
+                "streaming": True,
+            },
+        )
+
+    except Exception as e:
+        yield TaskOutput(
+            content=f"Custom blog analysis failed: {str(e)}",
+            metadata={
+                "error": True,
+                "function_name": "streaming_blog_analysis_function",
+                "streaming": True,
+            },
+        )
+
+
+def streaming_team_research_function(
```

Comment:
Maybe make this one simpler where the user just yields a string or something, as an example

Reply:
ok i will do these in the main PR when i redo and add more cookbooks